examples/dao/test/failing-path.js

```diff
@@ -0,0 +1,585 @@
+      assert.throws(
+        () => ctx.executeTx({ ...registerVoteParam, fromAccount: ctx.voterB.account }),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should fail user tries to register votes again for same proposal (double voting)', () => {
+      // register votes by voterA
+      ctx.executeTx({ ...registerVoteParam, fromAccount: ctx.voterA.account });
+
+      // fails (voter A tries to register votes again)
+      assert.throws(
+        () => ctx.executeTx({ ...registerVoteParam, fromAccount: ctx.voterA.account }),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should fail if latest_timestamp is < votingStart (i.e voting is not open)', () => {
+      // set now < votingStart
+      ctx.runtime.setRoundAndTimestamp(10, now + 30);
+
+      // deposit votes by voterB
+      ctx.depositVoteToken(ctx.voterB, ctx.depositLsig, 8);
+
+      assert.throws(
+        () => ctx.executeTx({ ...registerVoteParam, fromAccount: ctx.voterB.account }),
+        RUNTIME_ERR1009
+      );
+    });
+  });
+
+  describe('Execute', function () {
+    function resetCtx () {
+      // set up context
+      setUpCtx();
+
+      // optIn's
+      ctx.optInToDAOApp(ctx.proposalLsig.address());
+      ctx.optInToDAOApp(ctx.voterA.address);
+      ctx.optInToDAOApp(ctx.voterB.address);
+
+      // add proposal
+      ctx.addProposal();
+    }
+
+    this.beforeEach(() => {
+      resetCtx();
+    });
+
+    let executeProposalTx;
+    this.beforeEach(() => {
+      executeProposalTx = [
+        {
+          type: types.TransactionType.CallApp,
+          sign: types.SignType.SecretKey,
+          fromAccount: ctx.proposer.account,
+          appID: ctx.daoAppID,
+          payFlags: { totalFee: 2000 },
+          appArgs: ['str:execute'],
+          accounts: [ctx.proposalLsig.address()]
+        },
+        // tx1 as per proposal instructions
+        {
+          type: types.TransactionType.TransferAlgo,
+          sign: types.SignType.LogicSignature,
+          fromAccountAddr: ctx.daoFundLsig.address(),
+          toAccountAddr: ctx.proposer.address,
+          amountMicroAlgos: 2e6,
+          lsig: ctx.daoFundLsig,
+          payFlags: { totalFee: 0 } // fee must be paid by proposer
+        }
+      ];
+
+      // set current time after voting over
+      ctx.runtime.setRoundAndTimestamp(10, votingEnd + 10);
+    });
+
+    it('should reject execute if proposal is expired', () => {
+      // set current time as > executeBefore
+      ctx.runtime.setRoundAndTimestamp(10, executeBefore + 30);
+      assert.throws(
+        () => ctx.executeTx(executeProposalTx),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should reject execute if voting is still in progress', () => {
+      // set current time when "voting is active"
+      ctx.runtime.setRoundAndTimestamp(10, votingStart + 30);
+      assert.throws(
+        () => ctx.executeTx(executeProposalTx),
+        RUNTIME_ERR1009
+      );
+    });
```

Comment:
add: 
* execution should fail if now < votingStart (before voting started)

Reply:
done